pandas/core/categorical.py

```diff
@@ -1140,17 +1159,33 @@ def check_for_ordered(self, op):
                             "you can use .as_ordered() to change the "
                             "Categorical to an ordered one\n".format(op=op))
 
-    def argsort(self, ascending=True, **kwargs):
-        """ Implements ndarray.argsort.
-
-        For internal compatibility with numpy arrays.
+    def argsort(self, ascending=True, *args, **kwargs):
+        """
+        Returns the indices that would sort the Categorical instance,
+        provided the instance is ordered. This function is implemented
+        to provide compatibility with numpy ndarray objects.
 
         Only ordered Categoricals can be argsorted!
 
         Returns
         -------
         argsorted : numpy array
+
+        See also
+        --------
+        numpy.ndarray.argsort
         """
+
```

Comment:
I think I like better:

``ascending, kwargs = numpy_compat.argsort(self, ascending=ascending, *args, **kwargs)``

and all of the work is done in compat